genetic_algorithms/algorithms/algorithm.py

```diff
@@ -0,0 +1,54 @@
+from abc import abstractmethod
+from genetic_algorithms.helpers.camel_to_snake import camel_to_snake
+from typing import Union
+from genetic_algorithms.problems.base.model import Model
+from genetic_algorithms.problems.base.state import State
+from genetic_algorithms.models.next_state_provider import NextStateProvider
+from dataclasses import dataclass
+
+
+@dataclass
+class AlgorithmConfig:
+    max_steps_without_iprovement: int = 10
```

Comment:
To chyba literówka i powinno być "improvement"

Reply:
Dzięki!